src/composables/accountSelector.ts

```diff
@@ -0,0 +1,195 @@
+  PROTOCOLS,
+  TX_DIRECTION,
+} from '@/constants';
+import {
+  createCustomScopedComposable,
+  excludeFalsy,
+  getDefaultAccountLabel,
+  getDirection,
+} from '@/utils';
+import {
+  useAccounts,
+  useAddressBook,
+  useLatestTransactionList,
+} from '@/composables';
+
+import { useAeNames } from '@/protocols/aeternity/composables/aeNames';
+import { getInnerTransaction, getOwnershipStatus, getTxOwnerAddress } from '@/protocols/aeternity/helpers';
+
+import { tg } from '@/popup/plugins/i18n';
+
+export const useAccountSelector = createCustomScopedComposable(() => {
+  const { getName } = useAeNames();
+
+  const searchQuery = ref('');
+  const {
+    addressBookFiltered: rawAddressBookFiltered,
+    addressBookFilteredByProtocol,
+    protocolFilter,
+    showBookmarked,
+    getAddressBookEntryByAddress,
+    setProtocolFilter,
+    setShowBookmarked,
+    clearFilters: addressBookClearFilters,
+  } = useAddressBook();
+  const {
+    activeAccount,
+    accounts,
+    accountsGroupedByProtocol,
+    getAccountByAddress,
+  } = useAccounts();
+  const { accountsTransactionsLatest } = useLatestTransactionList();
+  const accountSelectType = ref<AccountSelectTypeFilter>(ACCOUNT_SELECT_TYPE_FILTER.addressBook);
+  const addressBookFiltered = computed(() => rawAddressBookFiltered.value.map((account) => ({
+    ...account,
+    isAddressBookEntry: true,
+  })));
+  const latestTransactions = computed(
+    () => (accountsTransactionsLatest.value[activeAccount.value.address] || [])
+      .map((transaction: ITransaction): IAccountSelectorEntry | null => {
+        const outerTx = transaction.tx!;
+        const innerTx = transaction.tx ? getInnerTransaction(transaction.tx) : null;
+        const { recipientId } = outerTx?.payerId ? outerTx : innerTx;
+        const direction = getDirection({
+          innerTx,
+          outerTx,
+          ownershipStatus: getOwnershipStatus(activeAccount.value, accounts.value, innerTx),
+          txOwnerAddress: getTxOwnerAddress(innerTx),
+          activeAccountAddress: activeAccount.value.address,
+        });
+        if (direction !== TX_DIRECTION.sent || !recipientId) {
+          return null;
+        }
+
+        let name = tg('modals.send.recipientLabel');
+        const isChainName = recipientId?.startsWith('nm_');
+        const addressBookRecord = getAddressBookEntryByAddress(recipientId);
+
+        if (addressBookRecord) {
+          return {
+            ...addressBookRecord,
+            isAddressBookEntry: true,
+          };
+        }
+
+        // Will be replaced in AccountInfo if it's a chain name
+        if (!isChainName) {
+          const account = getAccountByAddress(recipientId!);
+          if (account) {
+            name = getName(account.address).value || getDefaultAccountLabel(account);
+          } else {
+            name = getName(recipientId).value || name;
+          }
+        }
+
+        return {
+          name,
+          address: recipientId,
```

Comment:
I strongly suggest to avoid assigning non `ak_` addresses to the `account.address`. Previously we had this typed, so the `account` was forced to be `ak_`. With the introduction of multiple protocols we needed to change this to `string`, but still many places in the system requires the address to be `ak_`.

Reply:
Address should remain address, not a name hash.